### PR TITLE
Add extra debug assertions for WebGL errors

### PR DESCRIPTION
**Pull request by asajeffrey**
Add some extra debug assertions checking for GL errors, so we'll get a bit more help tracking down issues like #24788 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because

### CODE REVIEW

File: components/canvas/webgl_thread.rs [hunk 1]
```diff
@@ -680,10 +680,14 @@ impl WebGLThread {
             )
             .expect("Where's the GL data?");
 
+            // Ensure there are no pending GL errors from other parts of the pipeline.
```
Comment (Manishearth) [inline components/canvas/webgl_thread.rs]: Perhaps this comment could be a part of the assert string?

**Review by Manishearth — APPROVED** [review]
r=me, though if we can add assertion strings to the asserts that would be nice

**Comment by asajeffrey** [comment]
My experience is that assert strings are less use than the line numbers. YMMV. @bors-servo r=Manishearth

**Comment by jdm** [comment]
@bors-servo retry
* #25216

**Pull request merged**